public/New-DbaDbUser.ps1

```diff
@@ -88,136 +91,127 @@ function New-DbaDbUser {
+
+        Creates a new cointained sql user named user1 in the database given database with the password specified.
     #>
     [CmdletBinding(SupportsShouldProcess, ConfirmImpact = "Medium")]
     param(
-        [parameter(Mandatory, Position = 1)]
+        [parameter(Mandatory = $True, Position = 1)]
         [DbaInstanceParameter[]]$SqlInstance,
         [PSCredential]$SqlCredential,
         [string[]]$Database,
         [string[]]$ExcludeDatabase,
-        [switch]$IncludeSystem,
+        [switch]$IncludeSystem = $False,
+        [Parameter(ParameterSetName = "NoLogin", Mandatory = $False)]
+        [Parameter(ParameterSetName = "Login", Mandatory = $True)]
         [string]$Login,
-        [string]$Username = $Login,
+        [Parameter(ParameterSetName = "Login", Mandatory = $False)]
+        [Parameter(ParameterSetName = "NoLogin", Mandatory = $True)]
+        [Parameter(ParameterSetName = "ContainedSQLUser", Mandatory = $True)]
+        [Parameter(ParameterSetName = "ContainedAADUser", Mandatory = $True)]
+        [string]$UserName,
         [string]$DefaultSchema = 'dbo',
+        [Parameter(ParameterSetName = "ContainedSQLUser", Mandatory = $True)]
+        [securestring]$Password,
+        [Parameter(ParameterSetName = "ContainedAADUser", Mandatory = $True)]
         [switch]$ExternalProvider,
         [switch]$Force,
         [switch]$EnableException
     )
 
     begin {
+        $connParam = @{ }
+        if ($SqlCredential) { $connParam.SqlCredential = $SqlCredential }
+
+        # is this required?
         if ($Force) { $ConfirmPreference = 'none' }
 
-        function Remove-User {
-            param (
-                [Microsoft.SqlServer.Management.Smo.User]$User
-            )
-            if ($Force) {
-                if ($Pscmdlet.ShouldProcess($User, "Dropping existing user $($User.Name) in the database $($User.Parent.Name) on $instance because -Force was used")) {
-                    try {
-                        $User.Drop()
-                    } catch {
-                        Stop-Function -Message "Could not remove existing user $($User.Name) in the database $($User.Parent.Name) on $instance, skipping." -Target $User -ErrorRecord $_ -Continue
-                    }
-                }
-            } else {
-                Stop-Function -Message "User $($User.Name) already exists in the database $($User.Parent.Name) on $instance and -Force was not specified, skipping." -Target $User -Continue
-            }
+        # When user is created from login and no user name is provided then login name will be used as the user name
+        if ($Login -and -not($UserName)) {
+            $UserName = $Login
+        }
+
+        #Set appropriate user type
+        #Removed SQLLogin  user type. This is deprecated and the alternate is to map to SqlUser. Reference https://learn.microsoft.com/en-us/dotnet/api/microsoft.sqlserver.management.smo.usertype?view=sql-smo-160
```

Comment:
Your link does not show that is deprecated. This will still work on lower versions of SQL Server?

Reply:
It is mentioned in the fields section.  https://learn.microsoft.com/en-us/dotnet/api/microsoft.sqlserver.management.smo.usertype?view=sql-smo-160#fields

`SqlLogin	0	Specifies a SQLLogin user. This is deprecated, use SQLUser instead.`

I might be wrong. I thought the smo library is part of `dbatools` or `dbatools.library` module as a dependency which should be the recent version. If this is the case then I will prefer to keep the changes. However, if the module uses smo library installed locally then it makes sense to revert it back. I was not clear, your inputs will be helpful.